apps/system_apps/systemctl/main.py

```diff
@@ -41,29 +41,28 @@ def change_filters():
         checkbox_contents.append([type[0], type[1], type[1] in config["allowed_types"]])
     states = Checkbox(checkbox_contents, i, o, final_button_name="Save").activate()
     if states is None: return None
-    config["allowed_types"] = [state for state in states if states[state]] 
+    config["allowed_types"] = [state for state in states if states[state]]
     write_config(config, config_path)
-    
+
 def all_units():
     menu_contents = []
     units = systemctl.list_units()
-    for unit in units: 
+    for unit in units:
         menu_contents.append([unit["basename"], lambda x=unit: unit_menu(x)])
     Menu(menu_contents, i, o, "Systemctl: all unit list menu").activate()
 
 def pinned_units():
     menu_contents = []
-    units = systemctl.list_units()
-    for unit_name in config["pinned_units"]:
-        menu_contents.append([unit_name, lambda x=unit_name: unit_menu({"name":x}, in_pinned=True)])
+    units = systemctl.list_units("name", config["pinned_units"])
+    for unit in units:
+        menu_contents.append([unit["name"], lambda x=unit: unit_menu(x)])
```

Comment:
You've lost the `in_pinned=True`, as a result, a pinned unit can't be unpinned from the "Pinned units" menu